index.js

```diff
@@ -190,16 +196,34 @@ module.exports = function (schema, options) {
                     schema.statics[method + 'Deleted'] = function () {
                         var args = [];
                         Array.prototype.push.apply(args, arguments);
+                        // Handle session in options (usually last argument)
+                        var options = {};
+                        var lastArg = arguments[arguments.length - 1];
+                        if (lastArg && typeof lastArg === 'object' && lastArg.session) {
+                            options.session = lastArg.session;
+                        }
                         var match = { $match : { deleted : {'$ne': false } } };
                         arguments.length ? args[0].unshift(match) : args.push([match]);
+                         // Pass the session in the options if present
+                        if (options.session) {
+                            return Model[method].apply(this, args).session(options.session);
+                        }
```

Comment:
🤏 This would be a little cleaner if you didn't store the session in the options object

```suggestion
                        // Handle session in options (usually last argument)
                        var session;
                        var lastArg = arguments[arguments.length - 1];
                        if (lastArg && typeof lastArg === 'object' && lastArg.session) {
                            session = lastArg.session;
                        }
                        var match = { $match : { deleted : {'$ne': false } } };
                        arguments.length ? args[0].unshift(match) : args.push([match]);
                         // Pass the session in the options if present
                        if (session) {
                            return Model[method].apply(this, args).session(session);
                        }
```